crates/polars-ops/src/series/ops/ewm_by.rs

```diff
@@ -1,23 +1,29 @@
+use arrow::compute::concatenate::concatenate_validities;
+use arrow::compute::utils::combine_validities_and;
+use bytemuck::allocation::zeroed_vec;
 use num_traits::{Float, FromPrimitive, One, Zero};
 use polars_core::prelude::*;
+use polars_core::utils::align_chunks_binary;
 
 pub fn ewm_mean_by(
     s: &Series,
     times: &Series,
     half_life: i64,
     assume_sorted: bool,
 ) -> PolarsResult<Series> {
-    let func = match assume_sorted {
-        true => ewm_mean_by_impl_sorted,
-        false => ewm_mean_by_impl,
-    };
     match (s.dtype(), times.dtype()) {
-        (DataType::Float64, DataType::Int64) => {
-            Ok(func(s.f64().unwrap(), times.i64().unwrap(), half_life).into_series())
-        },
-        (DataType::Float32, DataType::Int64) => {
-            Ok(ewm_mean_by_impl(s.f32().unwrap(), times.i64().unwrap(), half_life).into_series())
-        },
+        (DataType::Float64, DataType::Int64) => Ok((if assume_sorted {
+            ewm_mean_by_impl_sorted(s.f64().unwrap(), times.i64().unwrap(), half_life)
+        } else {
+            ewm_mean_by_impl(s.f64().unwrap(), times.i64().unwrap(), half_life)
+        })
+        .into_series()),
+        (DataType::Float32, DataType::Int64) => Ok((if assume_sorted {
+            ewm_mean_by_impl_sorted(s.f32().unwrap(), times.i64().unwrap(), half_life)
+        } else {
+            ewm_mean_by_impl(s.f32().unwrap(), times.i64().unwrap(), half_life)
+        })
+        .into_series()),
```

Comment:
drive-by, but `ewm_mean_by_impl_sorted` wasn't being used for `DataType::Float32`